### ISSUE TITLE
Using on non-composer project ?

### ISSUE THREAD

**Issue by beppe9000** [vimeo/psalm#5024]
can i use the phar to specify a folder?

**Comment by weirdan** [comment]
Yes.

**Comment by beppe9000** [comment]
how? it does complain about composer autoloader missing

**Comment by orklah** [comment]
Do you have your own autoloader? Or do you proceed only with include?

**Comment by orklah** [comment]
Ok.

Can you try adding 
```xml
<extraFiles>
    <directory name="your_source" />
</extraFiles>
```

in your xml config just below `<projectFiles>`?

This should help Psalm know there is code it can use here.

**Comment by weirdan** [comment]
> This should help Psalm know there is code it can use here.

Note though that *functions* (as opposed to *classes*) found there won't be made available globally. This is something I intend to look into eventually as I have a legacy codebase that would benefit from it.

**Comment by orklah** [comment]
I think adding requiring your functions through the custom autoloader should work in that case. Whatever function added at this stage will be available

**Comment by weirdan** [comment]
> through the custom autoloader 

You mean the one specified in `psalm.xml`? I really don't want to load those files into the PHP process executing Psalm. Also, the files containing them may have side effects.

**Comment by orklah** [comment]
Yeah, that's what I meant, but indeed, those files would be executed as a result.

**Comment by beppe9000** [comment]
so basically the global functions are ignored unless I require them one by one in a project-level autoloader, right ?

**Comment by beppe9000** [comment]
Anyway I created the xml file manually and it does do a normal scan, but if I switch to taint analysys it does crash because of missing imports
